packages/core/src/controllers/ConnectorController.ts

```diff
@@ -222,5 +220,45 @@ export const ConnectorController = {
       themeVariables,
       w3mThemeVariables: getW3mThemeVariables(themeVariables, themeMode)
     })
+  },
+
+  /**
+   * Returns the connectors filtered by namespace.
+   * @param namespace - The namespace to filter the connectors by.
+   * @returns ConnectorWithProviders[].
+   */
+  getConnectorsByNamespace(namespace: ChainNamespace) {
+    const namespaceConnectors = state.allConnectors.filter(
+      connector => connector.chain === namespace
+    )
+
+    return this.mergeMultiChainConnectors(namespaceConnectors)
+  },
+
+  /**
+   * Returns the connectors. If a namespace is provided, the connectors are filtered by namespace.
+   * @param namespace - The namespace to filter the connectors by. If not provided, all connectors are returned.
+   * @returns ConnectorWithProviders[].
+   */
+  getConnectors(namespace?: ChainNamespace) {
+    if (namespace) {
+      return this.getConnectorsByNamespace(namespace)
+    }
+
+    return this.mergeMultiChainConnectors(state.allConnectors)
+  },
+
+  /**
+   * Sets the filter by namespace and updates the connectors.
+   * @param namespace - The namespace to filter the connectors by.
+   */
+  setFilterByNamespace(namespace: ChainNamespace) {
+    state.filterByNamespace = namespace
+    state.connectors = this.getConnectors(namespace)
+  },
+
+  clearNamespaceFilter() {
+    state.filterByNamespace = undefined
+    state.connectors = this.getConnectors()
   }
```

Comment:
why do we need this as a state field? 
Can't we have a getter by namespace instead of having to maintain a new piece of state?